py-polars/polars/io/parquet/functions.py

```diff
@@ -467,6 +476,10 @@ def scan_parquet(
         msg = "The `hive_schema` parameter of `scan_parquet` is considered unstable."
         issue_unstable_warning(msg)
 
+    if credential_provider is not None:
```

Comment:
Can we if the `credential_provider` and `storage_options` are `None`, the link is a cloud url and boto3/gcp is installed, insert a `credential_provider` function on behalf of the user?

Reply:
We could, but then I think there wouldn't be any way for users to disable that behavior, so I'm not sure.

I'm thinking in a follow-up PR we can add something that lets users do

```python
pl.scan_parquet(..., credential_provider=pl.CredentialProviderAWS(sts_role=..., profile_name=...))
```

Reply:
But ok, let's do that in a follow up PR.